packages/fuels-signers/src/provider.rs

```diff
@@ -410,4 +410,30 @@ impl Provider {
             })
             .unwrap_or(0)
     }
+
+    pub async fn get_messages(&self, from: &Bech32Address) -> Result<Vec<Message>, ProviderError> {
+        let mut messages: Vec<Message> = vec![];
+
+        let mut cursor = None;
+
+        loop {
+            let pagination = PaginationRequest {
+                cursor: cursor.clone(),
+                results: 9999,
```

Comment:
This isn't a very friendly query, typically pages of results shouldn't exceed 100 imo. I'd recommend making this a constant.